### DISCUSSION TITLE
GitHub REST API `app/installation-requests` seemingly inaccessible to all authentication methods

### DISCUSSION THREAD

**Discussion by tihal** [community/community#61877] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello! For context, I'm working on an installation flow where a user can install our GitHub app to their organization, and we then save a copy of that installation ID in our database.

I'm tackling the user flow where a user will request installation of the GitHub app to their organization via an organization admin, and I would like to use the endpoint `/app/installation-requests` to get a view of any "pending" installs so we can match up users, teams, and install IDs once the admin approves the installation.

[API reference here.](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#list-installation-requests-for-the-authenticated-app)

However, every time I try to access this using the Octokit SDK or via cURL, I receive either a 401 error, or an error stating `Error: [@octokit/auth-app] installationId option is required for installation authentication.`

I've tried using the documented method of instantiating a new Octokit instance with an Oauth GitHub user token and making the request using the .request method - that returns a 401 error.

I've tried authenticating as our App, authenticating on behalf of a user, and even authenticating as an Installation, even though you (in theory) shouldn't have to authenticate as an installation since an install ID isn't generated until after the installation is approved. None of these authentication methods allow me to access this endpoint.

Can anyone point me in the right direction on this or even recreate the issue? I'd be extremely grateful!

**Comment by jer-k** [comment]
First off, thank you for asking the question, somehow it was the only thing that popped up for me in searching for pending app installation requests.

Secondly, you should be able to do this using a `bearer_token`. I'm going to link some Ruby / Octokit.rb code but this appears to work.

```ruby
      def bearer_client
        Octokit::Client.new(bearer_token: jwt)
      end

      def jwt
        current = Time.current.to_f.floor
        key = {
          iat: current,
          exp: current + (10 * 60) - 30,
          iss: ENV["GITHUB_APP_CLIENT_ID"],
        }
        JWT.encode(key, github_app_private_key, "RS256")
      end

      def github_app_private_key
        OpenSSL::PKey::RSA.new(ENV["GITHUB_APP_PEM_KEY"])
      end
```

Now I can do

```ruby
bearer_client.get("/app/installation-requests")
=> [
  {
    :id=> 123,
    :node_id=>"asdf",
    :account=>
    {:login=>"jer-k",
    ...
  }
]
```
